lib/session.ex

```diff
@@ -0,0 +1,198 @@
+defmodule Janus.Session do
+  @moduledoc """
+  Keeps information about session and sends keep-alive messages
+  if given connection's transport requires it.
+  """
+
+  use GenServer
+  alias Janus.Connection
+
+  @default_timeout 5000
+
+  @type t :: pid()
+  @type message_t :: map()
+  @type connection_t :: pid()
+  @type timeout_t :: non_neg_integer()
+  @type session_id_t :: non_neg_integer()
+  @type plugin_t :: String.t()
+
+  @type plugin_handle_id :: pos_integer
+
+
+
+  @doc """
+  Synchronously creates a new session on the gateway.
+
+  ## Arguments
+
+  * `connection` - a PID of the `Janus.Connection` process,
+  * `timeout` - a timeout for the call.
+
+  ## Return values
+
+  On success it returns `{:ok, session}` where session is
+  a pid of `Janus.Session` process which keeps track of
+  session indentifier used by the gateway. Session process is linked with
+  calling process.
+
+  On error it returns `{:error, reason}`.
+
+  The reason might be:
+
+  * `{:gateway, code, info}` - it means that the call itself succeded but the
+    gateway returned an error of the given code and info,
+  * other - some serious error happened.
+  """
+  @spec create_linked_session(connection_t(), timeout_t()) :: {:ok, Janus.Session.t()} | {:error, any}
+  def create_linked_session(connection, timeout \\ @default_timeout) do
+    case Connection.call(connection, %{janus: :create}, timeout) do
+      {:ok, %{"id" => session_id}} ->
+        start_link(session_id, connection)
+
+      # TODO: handle positive errors
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  @spec start_link(session_id_t(), connection_t()) :: GenServer.server()
+  def start_link(session_id, connection) do
+    GenServer.start_link(__MODULE__, {session_id, connection}, [])
+  end
+
+  @doc """
+  Synchronously attaches a plugin to the session on the gateway.
```

Comment:
Janus describes it differently:
```suggestion
  Synchronously attaches to a plugin via session on the gateway.
```